test/layouts/coordinates.cpp

```diff
@@ -124,6 +124,32 @@ TEST_CASE("siqad coordinate conversion", "[coordinates]")
     CHECK(t5_fiction.y == -3);
 }
 
+TEST_CASE("Offset coordinate iteration", "[coordinates]")
+{
+    using lyt = cartesian_layout<offset::ucoord_t>;
+
+    std::stringstream print_stream;
+
+    lyt{{1, 1, 1}}.foreach_coordinate([&ps = print_stream](const auto& c) { ps << c.str(); });
+
+    constexpr const char* coordinates_string = "(0,0,0)(1,0,0)(0,1,0)(1,1,0)(0,0,1)(1,0,1)(0,1,1)(1,1,1)";
+
+    CHECK(coordinates_string == print_stream.str());
+}
+
+TEST_CASE("SiQAD coordinate iteration", "[coordinates]")
+{
+    using lyt = cartesian_layout<siqad::coord_t>;
+
+    std::stringstream print_stream;
+
+    lyt{{1, 1, 1}}.foreach_coordinate([&ps = print_stream](const auto& c) { ps << c.str(); });
+
+    constexpr const char* coordinates_string = "(0,0,0)(1,0,0)(0,0,1)(1,0,1)(0,1,0)(1,1,0)(0,1,1)(1,1,1)";
+
+    CHECK(coordinates_string == print_stream.str());
+}
```

Comment:
These checks work, but I would advise pushing the coordinates into an `std::vector` and then comparing it against a pre-defined one via `==`.

Reply:
Implemented in the latest commit